core/hash/compare_by_identity_spec.rb

```diff
@@ -44,10 +44,7 @@ def o.hash; 123; end
     :bar.should equal(:bar)
     @idh[:bar] = :e
     @idh[:bar] = :f
-    "bar".should_not equal('bar')
-    @idh["bar"] = :g
-    @idh["bar".dup] = :h
-    @idh.values.should == [:c, :d, :f, :g, :h]
+    @idh.values.should == [:c, :d, :f]
```

Comment:
Removed this part that would have been redundant of the first two elements if the string was wrapped into an array